crates/cli/src/lib.rs

```diff
@@ -1100,9 +1106,9 @@ fn roc_dev_native(
                         memory.reset();
                     }
                 }
-            }
+            };
 
-            std::process::exit(0)
+            std::process::exit(exit_code)
```

Comment:
@ayazhafiz I double-checked for you:
```
roc on  main [!] is 📦 v0.0.1 via 🦀 v1.71.1 via ❄️  impure (nix-shell-env) took 4s 
❯ ./target/release/roc dev /home/username/gitrepos/roc5/roc/crates/glue/tests/fixtures/nullable-wrapped/app.roc
🔨 Rebuilding platform...
tag_union was: StrFingerTree::More("foo", StrFingerTree::More("bar", StrFingerTree::Empty))
`More "small str" (Single "other str")` is: StrFingerTree::More("small str", StrFingerTree::Single("other str"))
`More "small str" Empty` is: StrFingerTree::More("small str", StrFingerTree::Empty)
`Single "small str"` is: StrFingerTree::Single("small str")
`Empty` is: StrFingerTree::Empty

roc on  main [!] is 📦 v0.0.1 via 🦀 v1.71.1 via ❄️  impure (nix-shell-env) 
❯ echo $?
0

roc on  main [!] is 📦 v0.0.1 via 🦀 v1.71.1 via ❄️  impure (nix-shell-env) 
❯ ./crates/glue/tests/fixtures/nullable-wrapped/app
tag_union was: StrFingerTree::More("foo", StrFingerTree::More("bar", StrFingerTree::Empty))
`More "small str" (Single "other str")` is: StrFingerTree::More("small str", StrFingerTree::Single("other str"))
`More "small str" Empty` is: StrFingerTree::More("small str", StrFingerTree::Empty)
`Single "small str"` is: StrFingerTree::Single("small str")
`Empty` is: StrFingerTree::Empty
Segmentation fault (core dumped)
```
It is segfaulting on main but it's hidden because the output was correct. We used to forget to check the exit code (which indicated the segfault) this PR fixes that.